More documentation (#116)
d3ledger/d3-btc: btc-withdrawal/README.md

## Bitcoin withdrawal service

> The Bitcoin withdrawal service(or simply `btc-withdrawal`) is a service dedicated to creating and signing Bitcoin withdrawal transactions.

Reviewer: again – the

### Simplified flow
1) The service listens to transfer commands in Iroha blockchain

> 2) If a transfer command is valid (destination address is whitelisted and it has a `base58` format), then the service tries to establish 'withdrawal consensus' among all the nodes. They must decide what UTXO may be spent.

Reviewer: all nodes or all of the nodes

3) Once the 'withdrawal consensus' is reached, the service starts creating a Bitcoin withdrawal transaction.

> 4) The next step is signing. All the nodes must sign a newly created transaction.

Reviewer: all or all of the nodes

> 5) If all the nodes(or the majority of them) signed a Bitcoin withdrawal transaction properly, it will be sent to the Bitcoin network.

Reviewer: odes(or  - no space

Assets will be rolled back to the initiator of transfer in case of error or failure.    
### Configuration overview (withdrawal.properties)
* `btc-withdrawal.withdrawalCredential` - D3 clients send assets to this account in order to execute withdrawal. The service reacts to the account's transfers and starts the withdrawal process. The account is also used to perform rollbacks. This account must be multisignature.
* `btc-withdrawal.signatureCollectorCredential` - this account is used to collect Bitcoin withdrawal transaction signatures. The account creates other accounts named after transaction hashes in `btcSignCollect` domain and saves signatures in it.
* `btc-withdrawal.btcConsensusCredential` - this account is used to create 'withdrawal consensus'.
* `btc-withdrawal.registrationCredential` - the account that was used to register D3 clients in the Bitcoin network. This account is needed to get clients whitelists.
* `btc-withdrawal.changeAddressesStorageAccount` - Iroha account that we use to store Bitcoin change addresses.

> * `btc-withdrawal.irohaBlockQueue` - name of a RabbitMQ queue that will be used to read Iroha blocks from

Reviewer 2: ```suggestion
* `btc-withdrawal.irohaBlockQueue` - name of the RabbitMQ queue to read Iroha blocks from
```
